logprep/util/helper.py

```diff
@@ -124,9 +131,11 @@ def get_dotted_field_value(event: dict, dotted_field: str) -> Optional[Union[dic
 
     fields = [event, *dotted_field.split(".")]
     try:
-        return reduce(dict.__getitem__, fields)
+        return reduce(_get_item, fields)
     except KeyError:
         return None
+    except ValueError:
+        return None
     except TypeError:
         return None
```

Comment:
Here you also have to check for the `IndexError` and write in the `test_helper.py` file a corresponding test. Just for the cases where someone tries to access an element, that is outside of the boundaries of the collection.